From 289006091fcc306343e1316b4e4a31a732760bbb Mon Sep 17 00:00:00 2001
From: Dinolek <git@dinolek.me>
Date: Sat, 12 Jul 2025 18:12:49 +0200
Subject: [PATCH 099/115] arm64: dts: mediatek: mt6785-xiaomi-begonia: Enable
 PMIC keys

Enable MT6359 PMIC keys and define power key.

Signed-off-by: Dinolek <git@dinolek.me>
---
 arch/arm64/boot/dts/mediatek/mt6785-xiaomi-begonia.dts | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/arch/arm64/boot/dts/mediatek/mt6785-xiaomi-begonia.dts b/arch/arm64/boot/dts/mediatek/mt6785-xiaomi-begonia.dts
index d12617f10..29c5d632b 100644
--- a/arch/arm64/boot/dts/mediatek/mt6785-xiaomi-begonia.dts
+++ b/arch/arm64/boot/dts/mediatek/mt6785-xiaomi-begonia.dts
@@ -137,4 +137,13 @@ pins-volume {
 
 &pmic {
 	interrupts-extended = <&pio 193 IRQ_TYPE_LEVEL_HIGH>;
+
+	keys {
+		compatible = "mediatek,mt6359-keys";
+
+		power {
+			linux,keycodes = <KEY_POWER>;
+			wakeup-source;
+		};
+	};
 };
-- 
2.56.0

